### PR TITLE
Add Socket icon

### PR DESCRIPTION
**Pull request by joshpinto6**
socketdotdev(1)

**Popularity metric:** According to [Similar Web](https://www.similarweb.com/website/socket.dev), Socket.dev has a global rank of `#330,152`, which is less than 500,000.

### Checklist

- [x] I updated the JSON data in `_data/simple-icons.json`
- [x] I optimized the icon with SVGO or SVGOMG
- [x] The SVG `viewbox` is `0 0 24 24`

### Description

- I picked the hex value #C93CD7 as it is directly between the two gradient values in Socket's main logo. Whileas the socket website uses a different color for its navigation bar and chakra UI styles, the logo seems to always use this gradient.
- I used the SVGO online to process my SVG and edited it in Affinity Designer.
- Additionally, because shields.io uses Simple Icons as a source by default and Socket is becoming a prominent tool in codebase security management, having it added to Simple Icons will make it much easier for people to create and use repository README badges for Socket reports.

### CODE REVIEW

File: _data/simple-icons.json [hunk 1]
```diff
@@ -14835,6 +14835,17 @@
             "hex": "000000",
             "source": "https://society6.com"
         },
+        {
+            "title": "Socket.dev",
```
Comment (LitoMore) [inline _data/simple-icons.json:14839]: They call themselves `Socket`. Let's use `Socket` as title instead.

File: _data/simple-icons.json [hunk 2]
```diff
@@ -14835,6 +14835,17 @@
...
+            "hex": "C93CD7",
+            "source": "https://socket.dev",
+            "aliases": {
+                "aka": [
+                    "Socket",
+                    "Socket Security"
+                ]
+            }
```
Comment (LitoMore) [inline _data/simple-icons.json:14847]: These aliases are not well-known aliases. Please consider removing them.

File: _data/simple-icons.json [hunk 3]
```diff
@@ -14835,6 +14835,17 @@
             "hex": "000000",
             "source": "https://society6.com"
         },
+        {
+            "title": "Socket.dev",
+            "hex": "C93CD7",
+            "source": "https://socket.dev",
```
Comment (LitoMore) [inline _data/simple-icons.json:14841]: Where did you get the SVG source file? I only see the PNG logo on the homepage.

We need the SVG source file for reviewing purposes.

**Review by LitoMore — CHANGES_REQUESTED** [review]

File: _data/simple-icons.json (hunk 1 above)
Comment (joshpinto6) [reply _data/simple-icons.json]: That's probably better. I made this change.

File: _data/simple-icons.json (hunk 2 above)
Comment (joshpinto6) [reply _data/simple-icons.json]: I removed them.

File: _data/simple-icons.json (hunk 3 above)
Comment (joshpinto6) [reply _data/simple-icons.json]: I updated the source field to reflect the official socketDev GitHub repository which I found it in:
https://github.com/SocketDev/vscode-socket-security/blob/master/logo-red.svg

**Review by LitoMore — APPROVED** [review]

**Comment by LitoMore** [comment]
@joshpinto6 Congrats on your first contribution to Simple Icons! Thanks!

**Pull request merged**